src/mixins/ExecutorMixin.h

```diff
@@ -19,12 +19,43 @@
 
 namespace reactivesocket {
 
+folly::Executor& defaultExecutor();
+
+class ExecutorBase {
+ public:
+  explicit ExecutorBase(
+      folly::Executor& executor = defaultExecutor(),
+      bool startExecutor = true);
+
+  /// We start in a queueing mode, where it merely queues signal
+  /// deliveries until ::start is invoked.
+  ///
+  /// Calling into this method may deliver all enqueued signals immediately.
+  void start();
+
+ protected:
+  template <typename F>
+  void runInExecutor(F&& func) {
+    if (pendingSignals_) {
+      pendingSignals_->emplace_back(func);
+    } else {
+      executor_.add(std::move(func));
+    }
+  }
+
+ private:
+  using PendingSignals = std::vector<std::function<void()>>;
+  std::unique_ptr<PendingSignals> pendingSignals_;
```

Comment:
this is because we are differentiating a case when pendingSignals_ is null or pendingSignals_ is empty. Please, see method start().
If we wanted to save the allocation, we would add a boolean value signaling whether we are still just accumulating user signals or executing them right away.